pep-0701.rst

```diff
@@ -0,0 +1,366 @@
+=============
+
+The formal proposed PEG grammar specification for f-strings is (see :pep:`617`
+for details on the syntax):
+
+.. code-block:: peg
+
+    fstring
+        | FSTRING_START fstring_middle* FSTRING_END
+    fstring_middle
+        | fstring_replacement_field
+        | FSTRING_MIDDLE
+    fstring_replacement_field
+        | '{' (yield_expr | star_expressions) "="? [ "!" NAME ] [ ':' fstring_format_spec* ] '}'
+    fstring_format_spec:
+        | FSTRING_MIDDLE
+        | fstring_replacement_field
+
+This PEP leaves up to the implementation the level of f-string nesting allowed.
+This means that limiting nesting is **not part of the language specification**
+but also the language specification **doesn't mandate arbitrary nesting**. 
+
+Three new tokens are introduced:
+
+* ``FSTRING_START``: This token includes f-string character (``f``/``F``) and the open quote(s).
+* ``FSTRING_MIDDLE``: This token includes everything between two expression braces (``}`` and ``{``).
+* ``FSTRING_END``: This token includes everything after the last expression brace (or the whole literal part
+  if no expression exists) until the closing quote.
+
+These tokens are always string parts and they are semantically equivalent to the
+``STRING`` token with the restrictions specified. These tokens must be produced by the lexer
+when lexing f-strings.  This means that **the tokenizer cannot produce a single token for f-strings anymore**. How
+the lexer emits this token is **not specified** as this will heavily depend on every
+implementation (even the Python version of the lexer in the standard library is
+implemented differently to the one used by the PEG parser).
+
+As an example::
+
+    f'some words {a+b} more words {c+d} final words'
+
+will be tokenized as::
+
+    FSTRING_START - "f'"
+    FSTRING_MIDDLE - 'some words '
+    LBRACE - '{'
+    NAME - 'a'
+    PLUS - '+'
+    NAME - 'b'
+    RBRACE - '}'
+    FSTRING_MIDDLE - ' more words '
+    LBRACE - '{'
+    NAME - 'c'
+    PLUS - '+'
+    NAME - 'd'
+    RBRACE - '}'
+    FSTRING_END - ' final words' (without the end quote)
+
+while ``f"""some words"""`` will be tokenized simply as::
+
+    FSTRING_START - 'f"""'
+    FSTRING_END - 'some words'
+
+All restrictions mentioned in the PEP are lifted from f-literals, as explained below:
+
+* Expression portions may now contain strings delimited with the same kind of
+  quote that is used to delimit the f-literal.
+* Backslashes may now appear within expressions just like anywhere else in
+  Python code. In case of strings nested within f-literals, escape sequences are
+  expanded when the innermost string is evaluated.
+* Comments, using the ``#`` character, are possible only in multi-line f-literals,
+  since comments are terminated by the end of the line (which makes closing a
+  single-line f-literal impossible)
+
+Backwards Compatibility
+=======================
+
+This PEP is not backwards incompatible: any valid Python code will continue to
```

Comment:
```suggestion
This PEP is backwards compatible: any valid Python code will continue to
```

a slight suggestion, instead of the double negation, had to do a double take when first reading it.

Reply:
Definitely agreed :+1: